packages/devtools_app/lib/src/profiler/cpu_profile_controller.dart

```diff
@@ -109,9 +126,32 @@ class CpuProfilerController with SearchControllerMixin<CpuStackFrame> {
     return matches;
   }
 
-  void loadData(CpuProfileData data) {
+  void loadProcessedData(CpuProfileData data) {
     assert(data.processed);
     _dataNotifier.value = data;
+    _dataByTag[userTagNone] = data;
+  }
+
+  Future<void> loadDataWithTag(String tag) async {
+    _userTagFilter.value = tag;
+
+    _dataNotifier.value = null;
+    _processingNotifier.value = true;
+
+    _dataNotifier.value = await processDataForTag(tag);
```

Comment:
wrap with a try+finally block in case an exception is thrown. otherwise you can get stuck processing.

Reply:
done. also added code to push a notification when the filter attempt fails.